src/PackageServiceProvider.php

```diff
@@ -213,4 +198,56 @@ public function packageView(?string $namespace)
     {
         return is_null($namespace) ? $this->package->shortName() : $this->package->viewNamespace;
     }
+
+    protected function bootPackageMigrations(): void
+    {
+        $now = Carbon::now();
+
+        if ($this->package->discoversMigrations) {
+            $this->discoverMigrations();
+
+            return;
+        }
+
+        foreach ($this->package->migrationFileNames as $migrationFileName) {
+            $filePath = $this->package->basePath("/../database/migrations/{$migrationFileName}.php");
+            if (! file_exists($filePath)) {
+                // Support for the .stub file extension
+                $filePath .= '.stub';
+            }
+
+            $this->publishes([
+                $filePath => $this->generateMigrationName(
+                    $migrationFileName,
+                    $now->addSecond()
+                ), ], "{$this->package->shortName()}-migrations");
+
+            if ($this->package->runsMigrations) {
+                $this->loadMigrationsFrom($filePath);
+            }
+        }
+    }
+
+    protected function discoverMigrations(): void
+    {
+        $now = Carbon::now();
+
+        $files = (new Filesystem())->files($this->package->basePath('/../database/migrations'));
```

Comment:
What happens if this path does not exist?

Reply:
Symfony throws a directory not found exception.

I'll allow package developers to define a `$path` parameter for `discoversMigrations` and default it to `'/../database/migrations'` as @erikgaal